*** inflate.c   Sun Oct  3 19:33:51 2004
--- inflate-1.2.2p1.c   Tue May 24 21:31:00 2005
***************
*** 1263,1270 ****
      /* copy state */
      *dest = *source;
      *copy = *state;
!     copy->lencode = copy->codes + (state->lencode - state->codes);
!     copy->distcode = copy->codes + (state->distcode - state->codes);
      copy->next = copy->codes + (state->next - state->codes);
      if (window != Z_NULL)
          zmemcpy(window, state->window, 1U << state->wbits);
--- 1263,1274 ----
      /* copy state */
      *dest = *source;
      *copy = *state;
!     if (state->lencode >= state->codes &&
!         state->lencode <= state->codes + ENOUGH - 1)
!     {
!         copy->lencode = copy->codes + (state->lencode - state->codes);
!         copy->distcode = copy->codes + (state->distcode - state->codes);
!     }
      copy->next = copy->codes + (state->next - state->codes);
      if (window != Z_NULL)
          zmemcpy(window, state->window, 1U << state->wbits);
